variables.tf

```diff
@@ -127,8 +121,14 @@ variable "log_config_filter" {
   default     = "ALL"
 }
 
+variable "enable_dynamic_port_allocation" {
+  type        = bool
+  description = "Enable Dynamic Port Allocation. If minPorts is set, minPortsPerVm must be set to a power of two greater than or equal to 32."
+  default     = false
+
+}
 variable "enable_endpoint_independent_mapping" {
   type        = bool
   description = "Specifies if endpoint independent mapping is enabled."
-  default     = null
+  default     = true
```

Comment:
Any reason for changing this default?

Reply:
It is enabled by default as per the [terraform documentation](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_router_nat#enable_endpoint_independent_mapping) so I made it true here.

Reply:
This is not enabled by default in Google Cloud, see https://cloud.google.com/nat/docs/overview#specs-rfcs

> By default, Endpoint-Independent Mapping is disabled when you create a NAT gateway.

The terraform documentation should probably be updated to match GCP and this set to `false` as default.

Reply:
@jcalero I took a peek at the provider and it seems like they actually default to true. It might be worth opening an issue there to point out this difference. @nikhilmakhijani Could we keep this a null for now so we can follow the provider default?

https://github.com/hashicorp/terraform-provider-google/blob/53295dc35039ca8cd66da18cc758bf7487bd8c03/google/resource_compute_router_nat.go#L238-L243

Reply:
Yes, I have updated it to null for now. Can we check what will be the best practice then update it accordingly?

Reply:
It's worth noting that Endpoint-Independent Mapping (EIM) will be disabled by default in future Terraform provider versions in order to be in sync with the Google default setting:

https://github.com/hashicorp/terraform-provider-google/issues/10609

Speaking personally, I would definitely say the best practice for 99% of all deployments is to disable EIM but do enable Dynamic Port Allocation (DPA) as this will manage port allocations more efficiently and decrease the chance of Cloud NAT drops.

At a bare minimum, it should be an option to leverage DPA as the feature has been out over a year and I've yet to see any downside to using it.

Reply:
I believe when it is set to `null` it still ends up where is EIM is enabled. This is why I've opened a separate PR. https://github.com/terraform-google-modules/terraform-google-cloud-nat/pull/113